tests/test_async_client.py

```diff
@@ -0,0 +1,142 @@
+import unittest
+from unittest.mock import AsyncMock, patch
+import unittest.async_case
+import willofsteel
+from willofsteel.async_client import AsyncClient
+from willofsteel.exceptions import *
+from willofsteel.types import *
+
+
+class TestAsyncClient(unittest.async_case.IsolatedAsyncioTestCase):
+
+    def setUp(self):
+        patcher = patch(
+            'willofsteel.async_client.AsyncClient.request', new_callable=AsyncMock)
+        self.addCleanup(patcher.stop)
+        self.mock_request = patcher.start()
+        self.mock_request.return_value = {}, 200
+        self.client = AsyncClient('valid_key')
+
+    @patch('willofsteel.async_client.AsyncClient.request', new_callable=AsyncMock)
+    async def test_validate_key_valid_key_raises_no_exception(self, mock_get):
+        mock_get.return_value = {}, 200
+        try:
+            await self.client._verify_key()
+            self.assertTrue(True)
+        except InvalidKey:
+            self.fail("InvalidKey exception raised unexpectedly")
+
+    @patch('willofsteel.async_client.AsyncClient.request', new_callable=AsyncMock)
+    async def test_validate_key_invalid_key_raises_InvalidKey_exception(self, mock_get):
+        mock_get.return_value = {'detail': 'Not authenticated'}, 403
+        with self.assertRaises(willofsteel.exceptions.InvalidKey):
+            await self.client._verify_key()
+
+    @patch('willofsteel.async_client.AsyncClient.request', new_callable=AsyncMock)
+    async def test_get_player_returns_player_object(self, mock_get):
+        mock_get.return_value = {'user_id': 1,
```

Comment:
You shouldn't only have two fields as the response from the request, as it is always going to return data stated on our API documentation. You can obtain mock data/information about the data that is going to be returned [here](https://api.willofsteel.me/#tag/Player-Data/operation/get_player_data_player_get).

Even though this would return a Player object, that is not what we are testing.